zebra-network/src/peer_set/set.rs

```diff
@@ -479,14 +474,19 @@ where
                 }
 
                 // Unready -> Errored
-                Poll::Ready(Some(Err((key, UnreadyError::Inner(error))))) => {
+                Err((key, UnreadyError::Inner(error))) => {
                     debug!(%error, "service failed while unready, dropping service");
 
                     let cancel = self.cancel_handles.remove(&key);
                     assert!(cancel.is_some(), "missing cancel handle");
                 }
             }
         }
+
+        // There are no unready peers, so the unready_peers waker will never wake pending tasks.
+        *cx = Context::from_waker(original_waker);
```

Comment:
This line does nothing, because the task is the same, so the context and waker are the same:
https://github.com/ZcashFoundation/zebra/pull/7859/files#r1380722199